### PR TITLE
Feature/376 aws mmq

### LINKED ISSUE #376: Formalize AWS MMQ support
As a user
I want to be able to use the AWS MMQ implementation of RabbitMQ
In order that IAM and other AWS service management features can be used to unify the platform management of studioml dependencies

### PR DESCRIPTION
**Pull request by karlmutch**
### What does this do / why do we need it?

This add support for AWS MMQ RabbitMQ implementations.

### What should your reviewer look out for in this PR?

Look out for logging code that might leak secrets.

The bulk of this code is adding an explicit entry for specification of the RabbitMQ management server rather than just using a different port number on the same machine.

### Which issue(s) does this PR fix?

fixes #376

### CODE REVIEW

File: internal/runner/rmq.go [hunk 1]
```diff
@@ -343,38 +369,62 @@ var (
 	qCheck   sync.Once
 )
 
+func generateMgtRMQ(amqpURL string, amqpMgtURL string) (mgtURL amqp.URI, err kv.Error) {
+	if len(amqpURL) == 0 && len(amqpMgtURL) == 0 {
+		return mgtURL, kv.NewError("amqp-url, or amqp-mgt-url was not specified on the command line, or as an env var, cannot start rabbitMQ").With("stack", stack.Trace().TrimRuntime())
+	}
```
Comment (andreidenissov-cog) [inline internal/runner/rmq.go:375]: That's my lack of Golang knowledge, but what is mgtURL here? If it's named return value, is it initialized at this code point?

File: cmd/runner/rabbit.go [hunk 2]
```diff
@@ -28,21 +29,41 @@ var (
...
+			logger.Warn(kv.Wrap(errGo).With("url", *amqpURL).With("stack", stack.Trace().TrimRuntime()).Error())
+		}
+		if qURL.User != nil {
+			creds = qURL.User.String()
+		} else {
+			logger.Warn(kv.NewError("missing credentials in url").With("url", *amqpURL).With("stack", stack.Trace().TrimRuntime()).Error())
+		}
+		qURL.User = nil
+		queueURL = qURL.String()
 	}
```
Comment (andreidenissov-cog) [inline cmd/runner/rabbit.go:45]: Looks like the logic is the same for both URLs, why not factor it out into a function?

File: cmd/runner/rabbit.go [hunk 3]
```diff
@@ -28,21 +29,41 @@ var (
...
+		qMgtURL, errGo := url.Parse(os.ExpandEnv(*amqpMgtURL))
+		if errGo != nil {
+			logger.Warn(kv.Wrap(errGo).With("url", *amqpMgtURL).With("stack", stack.Trace().TrimRuntime()).Error())
+		}
+		if qMgtURL.User != nil {
+			creds = qMgtURL.User.String()
+		} else {
+			logger.Warn(kv.NewError("missing credentials in url").With("url", *amqpMgtURL).With("stack", stack.Trace().TrimRuntime()).Error())
+		}
+		qMgtURL.User = nil
```
Comment (andreidenissov-cog) [inline cmd/runner/rabbit.go:57]: So we are OK to show queues URLs here?

File: cmd/runner/rabbit.go [hunk 4]
```diff
@@ -28,21 +29,41 @@ var (
...
-	if qURL.User != nil {
-		creds = qURL.User.String()
-	} else {
-		logger.Warn(kv.NewError("missing credentials in url").With("url", *amqpURL).With("stack", stack.Trace().TrimRuntime()).Error())
+
+	if len(*amqpMgtURL) != 0 {
+		qMgtURL, errGo := url.Parse(os.ExpandEnv(*amqpMgtURL))
+		if errGo != nil {
+			logger.Warn(kv.Wrap(errGo).With("url", *amqpMgtURL).With("stack", stack.Trace().TrimRuntime()).Error())
+		}
```
Comment (andreidenissov-cog) [inline cmd/runner/rabbit.go:51]: If Parse() call goes bad, are we always able to continue after this warning?

File: internal/runner/rmq.go (hunk 1 above)
Comment (karlmutch) [reply internal/runner/rmq.go:375]: Yes it is a named return value. Returned values are initialized to their 'nil' value when the function is entered.

File: cmd/runner/rabbit.go (hunk 2 above)
Comment (karlmutch) [reply cmd/runner/rabbit.go:45]: It felt a little too small to break out.  I would end up with only a small reduction inside the top level function I think.  I could try it and see.

File: cmd/runner/rabbit.go (hunk 3 above)
Comment (karlmutch) [reply cmd/runner/rabbit.go:57]: Given the credentials were missing I felt it ok.  Of course there is a risk they would be just misparsed.  I could make a change to extract the host name and scheme and just show those.

Comment (andreidenissov-cog) [reply cmd/runner/rabbit.go:57]: Yeah, probably will be safer that way.

File: cmd/runner/rabbit.go (hunk 2 above)
Comment (karlmutch) [reply cmd/runner/rabbit.go:45]: It looks OK with the change I'll leave it in

File: cmd/runner/rabbit.go (hunk 4 above)
Comment (karlmutch) [reply cmd/runner/rabbit.go:51]: Yes I will with an empty result.  This then bails out later on when the connection is attempted, or is ignored.

**Review by andreidenissov-cog — APPROVED** [review]
Looks good to me.

**Pull request merged**